### PR TITLE
feat: add python documentation for functions, Auth, and Storage

### PR DESCRIPTION
**Pull request by J0**
## What kind of change does this PR introduce?

This PR contains the Python bindings and examples for
- Supabase Storage
- Supabase Auth (Partial)
- Supabase Functions

Note that this doesn't include the typedoc which is still being worked on. The rest of the Auth docs will be released once `gotrue-py` is moved to v2

## What is the current behavior?
No docs- partial Sphinx v1 docs on `gotrue-py` `storage-py` and `postgrest-py`

## What is the new behavior?

Docs for Storage, Auth (Partial) and Functions.

### CODE REVIEW

**Comment by alaister** [comment]
@dannykng, are you the best person to review this one? Cheers 🙏🏻

**Comment by MildTomato** [comment]
need to add a python [...slug] page for this.
I'll add in now

**Comment by J0** [comment]
Hey all,

Thanks for waiting - this PR is now ready for review. 
There's known issue with the sidebars which leads to `TypeScript Support` showing up as a sidebar entry but I'll be working with Fran to resolve that tomorrow.

Let us know if there's any feedback/comments
Joel

**Comment by J0** [comment]
Local preview for ref:

image

File: apps/docs/docs/ref/python/introduction.mdx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    <h2 className="text-base font-mono text-scale-1100">@supabase/supabase-py</h2>
+  </div>
+</div>
+
+<div className="max-w-xl">
+  This reference documents every object and method available in Supabase's isomorphic Python
+  library, supabase-py. You can use supabase-py to interact with your Postgres database, listen to
+  database changes, invoke Deno Edge Functions, build login and user management functionality, and
+  manage large files.
+</div>
```
Comment (dannykng) [inline apps/docs/docs/ref/python/introduction.mdx:20]: Placeholder for now—likely needs updates before merging.

**Comment by dannykng** [comment]
> Local preview for ref:
> 
> image

https://github.com/supabase/supabase/pull/10343/commits/439b09f825940172da6bf3b9585b9b86ad2a38da should fix all the code examples showing at the same time. 😅

**Comment by J0** [comment]
Thanks for the fix on the code examples! Forgot about those 😅

**Review by dannykng — APPROVED** [review]
@J0 do we have content for the Database and Release Notes sections? If not, we should hide these from the nav menu until we do. Looks good otherwise!

Screenshot_2023-01-19_at_10_51_37_AM

**Comment by J0** [comment]
Added in the Reference section and an entry for the database section. Next, I'm going to make FLUPs section by section so that we can get parity against the js lib

**Pull request merged**